cpp/src/gcylon/all2all/cudf_all_to_all.cpp

```diff
@@ -61,6 +65,30 @@ cylon::Status gcylon::net::AllToAll(const cudf::table_view &tv_with_parts,
   while (!all_to_all.isComplete()) {}
   all_to_all.close();
 
+  // if no table is received from any worker,
+  // create an empty table for those
+  for (int i = 0; i < received_tables.size(); ++i) {
+    if (received_tables[i] == nullptr) {
+      received_tables[i] = cudf::empty_like(tv);
+    }
+  }
+
+  return cylon::Status::OK();
+}
+
+cylon::Status gcylon::net::AllToAll(const cudf::table_view &tv,
+                                    const std::vector<cudf::size_type> &part_indices,
+                                    const std::shared_ptr<cylon::CylonContext> &ctx,
+                                    std::unique_ptr<cudf::table> &table_out) {
+
+  std::vector<std::unique_ptr<cudf::table>> received_tables;
+  received_tables.reserve(ctx->GetWorldSize());
+
+  RETURN_CYLON_STATUS_IF_FAILED(
+    gcylon::net::AllToAll(tv, part_indices, ctx, received_tables));
+
+  std::vector<cudf::table_view> views = gcylon::tablesToViews(received_tables);
```

Comment:
can we move this to `cudf::concatenate` method?

Reply:
cudf::concatenate is from cudf libarary.